### ISSUE TITLE
Automatically upgrade compute server image versions

### ISSUE THREAD

**Issue by novoselt** [sagemathinc/cocalc#7677]
I am not clear how exactly it happened and how I can reproduce it direclty:
- I had a deprovisioned server with GPU
- I turned off the GPU switch, which resulted in the image change to just "Python"
- I tried to start it, but got a message that this image with this version does not exist and I potentially need to use advanced settings
- I switched the image to something else, then back to "Python", in hopes that it will reset the version
- The server started without any issues

I think there is very little sense in trying to preserve image versions for deprovisioned servers, at least not when they were picked explicitly. Rather every time the server starts it should be "the last stable one".

**Comment by williamstein** [comment]
>  Rather every time the server starts it should be "the last stable one".

This is the case, unless you explicitly click the "Advanced" checkbox and select a specific image.    You can confirm this as an admin by clicking on settings and looking at the json at the bottom for "tag".  If that is not set, then the default latest tested available image is used.  If it is set, then that specific version is used.     Here's an example server I just created, where can see that the tag is NOT set:

<img width="626" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/6db2d616-d9cf-464b-918a-0f6c2200bac5">


There were a number of stupid cases of templates that I had created that had an explicit tag set in them.  A few days ago I went through and manually fixed all of those templates.  Perhaps your server had initially been made using one of those templates and had an explicit "advanced" tag set.

In short, this is a very real issue you're pointing out, but I recently fixed *one part of it* last weekend.

There are a number of servers "out there" that do have a specific tag set.  You can click "Advanced", then select the "Default" tag, to switch to always using the latest version:

<img width="856" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/cc92fb3c-efab-4cc7-b66c-5a86c85267f4">

The part that remains:

- when changing from one image to another, the tag should be cleared if it is set, so it's no longer valid.  E.g., in your case, you have some tag that only makes sense for a GPU image, and suddenly it's trying to use that tag for python, which makes no sense.  This is certainly a bug that I didn't know about, but also fixing it is probably 2 lines of code.

**Comment by novoselt** [comment]
Yes! I had Colab image which for some reason had an explicit version selected "by itself", i.e. I picked "Colab" without "Advanced", then clicked "Advanced" and 2024-07-01 was selected, not "Default". I switched back to "Python" with no version highlighted at all and trying to start it got an error message.
